thrift/thrift.bzl

```diff
@@ -2,11 +2,40 @@
 
 _thrift_filetype = FileType([".thrift"])
 
+def _common_prefix(strings):
+  pref = None
+  for s in strings:
+    if pref == None:
+      pref = s
+    elif s.startswith(pref):
+      pass
+    else:
+      tmp_pref = pref
+      for end in range(0, len(pref) + 1):
```

Comment:
(skylark may be a total language... No recursion, only for loops on collections, I think everything terminates).

If you consider any method error to be a type error (which a better compiler would have caught, I think it may be total....)